pandas/io/parsers.py

```diff
@@ -40,70 +48,79 @@ class ParserWarning(Warning):
 
 Parameters
 ----------
-filepath_or_buffer : string or file handle / StringIO
-    The string could be a URL. Valid URL schemes include
-    http, ftp, s3, and file. For file URLs, a
-    host is expected. For instance, a local file could be
+filepath_or_buffer : str, pathlib.Path, py._path.local.LocalPath or any object \
+with a read() method (such as a file handle or StringIO)
```

Comment:
Not sure if we want to include the read() method or just mention file handles and StringIO